### PR TITLE
unisonlang M4e

### PR DESCRIPTION
**Pull request by aslilac**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/unisonlang.rb [hunk 1]
```diff
@@ -21,23 +21,27 @@ class Unisonlang < Formula
...
-  depends_on arch: :x86_64
-
   uses_from_macos "xz" => :build
   uses_from_macos "zlib"
 
   resource "codebase-ui" do
-    url "https://github.com/unisonweb/codebase-ui/archive/refs/tags/release/M3.tar.gz"
-    sha256 "84be9135a821615f8fc73a0a894aa46a11c55393c43dc26e16a5ce75f8063012"
-    version "M3"
+    url "https://github.com/unisonweb/unison-local-ui/archive/4d196d5d775c36490527552b3d98c929fc178a37.tar.gz"
```
Comment (carlocab) [inline Formula/unisonlang.rb:28]: I thought versions here were supposed to match?

Comment (aslilac) [reply Formula/unisonlang.rb]: They hadn't tagged this repo yet, but they just tagged it this morning. Fixed now.

**Comment by aslilac** [comment]
I think there is a CI-long-timeout label available, yeah?

**Comment by carlocab** [comment]
Sorry, just used it on an older PR. We'll get to this one soon.

**Comment by carlocab** [comment]
10.15 and 11 have been running for about 15 hours, which probably shouldn't be happening. I've cancelled CI.

**Comment by aslilac** [comment]
Oh yikes, yeah something went wrong. I was expecting the slower machines to take ~3 hours at most.

**Comment by aslilac** [comment]
The logs from the macOS-11 and macOS-10.15 jobs are both basically empty. Is there any other way to get insight into what went wrong?

**Comment by SMillerDev** [comment]
Not really, if it's cancelled the logs are essentially lost

**Comment by aryairani** [comment]
How would one get the logs if it wasn't canceled? And can we do that?

**Comment by Bo98** [comment]
> How would one get the logs if it wasn't canceled? And can we do that?

Bottom of here: https://github.com/Homebrew/homebrew-core/actions/runs/2679866981

It hung on `npm run ui-core:install`. Unfortunately the command didn't output anything before doing so.

**Comment by aryairani** [comment]
Thanks @SMillerDev, @Bo98 Can we just re-run it to see if it was a fluke?

**Comment by carlocab** [comment]
Seems to have stalled again. All other runners finished in about an hour.

**Comment by cho-m** [comment]
Still failing at `npm run ui-core:install`, but seems inconsistent. We had a run where only Big Sur runner failed and another where only Catalina failed.

Should be getting stuck on `npx elm-git-install` since log output would exist if the command finished.

**Comment by carlocab** [comment]
Smells like a race

File: Formula/unisonlang.rb [hunk 2]
```diff
@@ -17,27 +17,41 @@ class Unisonlang < Formula
...
+      system "npm", "install", *Language::Node.local_npm_install_args
+
+      # Manually run commands from install script
+      # TODO: system "npm", "run", "ui-core:install"
+      system "npx", "--yes", "elm-git-install"
+      uicore = JSON.parse(File.read("elm-stuff/gitdeps/github.com/unisonweb/ui-core/elm.json"))
+      uicore["dependencies"].each do |name, version_range|
+        package = "#{name}@#{version_range.split(".")[0]}"
+        system "npx", "--yes", "elm-json", "install", package, "--yes"
+      end
```
Comment (cho-m) [inline Formula/unisonlang.rb:48]: Let's see if we can cause same issue by manually running commands and generate some extra output. May need to check if this is doing everything the same as script.

Comment (cho-m) [reply Formula/unisonlang.rb:48]: Sadly, not much more info.

For Big Sur failure, it got past `elm-git-install` but failed on first `elm-json install` without any output, e.g.
```console
❯ cat 02.npx
2022-09-14 03:27:16 +0000

npx
--yes
elm-git-install

github.com/unisonweb/ui-core => d6d9cd0

❯ cat 03.npx
2022-09-14 03:27:23 +0000

npx
--yes
elm-json
install
elm/browser@1
--yes

```

Other runners have logs like:

```console
❯ cat 02.npx
2022-09-14 03:26:11 +0000

npx
--yes
elm-git-install

github.com/unisonweb/ui-core => d6d9cd0

❯ cat 03.npx
2022-09-14 03:26:15 +0000

npx
--yes
elm-json
install
elm/browser@1
--yes

npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

-- PACKAGE CHANGES READY -------------------------------------------------------

Saved updated elm.json!

❯ cat 04.npx
2022-09-14 03:26:32 +0000

npx
--yes
elm-json
install
elm/core@1
--yes


-- PACKAGE CHANGES READY -------------------------------------------------------

Saved updated elm.json!
```

**Review by cho-m — APPROVED** [review]

**Comment by aslilac** [comment]
oh, nice! thanks for carrying this through!

**Pull request closed without merging**